javaslang-benchmark/src/test/java/javaslang/benchmark/collection/TreeMapBenchmark.java

```diff
@@ -0,0 +1,398 @@
+        }
+
+        @TearDown
+        public void tearDown() {
+            keysRandom = null;
+            keysRandom = null;
+            entriesRandom = null;
+            slangTreeMap = null;
+            javaTreeMap = null;
+            scalaTreeMap = null;
+        }
+
+        // Valid keys are only even numbers
+        int nextValidRandomKey() {
+            return keysRandom.get(random.nextInt(numItems));
+        }
+
+        int nextRandomKey() {
+            return random.nextInt(maxKey);
+        }
+
+        Array<Integer> randomize(Array<Integer> list) {
+            java.util.List<Integer> integers = list.toJavaList();
+            Collections.shuffle(integers);
+            return Array.ofAll(integers);
+        }
+    }
+
+    @Benchmark
+    @Group("GetAndPut")
+    public Object putRandomValues_Slang(BenchmarkState state) {
+        TreeMap<Integer, String> treeMap = TreeMap.empty();
+        for (Integer key : state.keysRandom) {
+            treeMap = treeMap.put(key, Integer.toString(key));
+        }
+        return treeMap;
+    }
+
+    @Benchmark
+    @Group("GetAndPut")
+    public Object putRandomValues_Java(BenchmarkState state) {
+        java.util.TreeMap<Integer, String> treeMap = new java.util.TreeMap<>();
+        for (Integer key : state.keysRandom) {
+            treeMap.put(key, Integer.toString(key));
+        }
+        return treeMap;
+    }
+
+    @Benchmark
+    @Group("GetAndPut")
+    public Object putRandomValues_Scala(BenchmarkState state) {
+        scala.collection.immutable.TreeMap<Integer, String> treeMap = new scala.collection.immutable.TreeMap<>(SCALA_ORDERING);
+        for (Integer key : state.keysRandom) {
+            treeMap = treeMap.$plus(new scala.Tuple2<>(key, Integer.toString(key)));
+        }
+        return treeMap;
+    }
+
+    @Benchmark
+    @Group("GetAndPut")
+    public Object putSortedValues_Slang(BenchmarkState state) {
+        TreeMap<Integer, String> treeMap = TreeMap.empty();
+        for (Integer key : state.keysSorted) {
+            treeMap = treeMap.put(key, Integer.toString(key));
+        }
+        return treeMap;
+    }
+
+    @Benchmark
+    @Group("GetAndPut")
+    public Object putSortedValues_Java(BenchmarkState state) {
+        java.util.TreeMap<Integer, String> treeMap = new java.util.TreeMap<>();
+        for (Integer key : state.keysSorted) {
+            treeMap.put(key, Integer.toString(key));
+        }
+        return treeMap;
+    }
+
+    @Benchmark
+    @Group("GetAndPut")
+    public Object putSortedValues_Scala(BenchmarkState state) {
+        scala.collection.immutable.TreeMap<Integer, String> treeMap = new scala.collection.immutable.TreeMap<>(SCALA_ORDERING);
+        for (Integer key : state.keysSorted) {
+            treeMap = treeMap.$plus(new scala.Tuple2<>(key, Integer.toString(key)));
+        }
+        return treeMap;
+    }
+
+    @Benchmark
+    @Group("GetAndPut")
+    public Object getValue_Slang(BenchmarkState state) {
```

Comment:
yeah, but that's my problem exactly, that libraries that will almost never be in the same classpath will end up being warmed up together.

I think @groups make our performance tests more unreliable and don't have any added value -- besides being quicker, apparently.

Please put validity checks in the performance tests also, so that we can check whether we're testing the same things (see my solution)